src/Http/Http/test/WritingAdaptersInteropTests.cs

```diff
@@ -0,0 +1,125 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Buffers;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace System.IO.Pipelines.Tests
+{
+    public class WritingAdaptersInteropTests : PipeStreamTest
+    {
+        [Fact]
+        public async Task CheckBasicWriteInterop()
+        {
+            var pipe = new Pipe();
+            var writeOnlyStream = new WriteOnlyPipeStream(pipe.Writer);
+            var pipeWriter = new StreamPipeWriter(writeOnlyStream);
+            await pipeWriter.WriteAsync(new byte[10]);
+
+            var res = await pipe.Reader.ReadAsync();
+            Assert.Equal(new byte[10], res.Buffer.ToArray());
+        }
+
+        [Fact]
+        public async Task CheckVeryNestedWriteInterop()
+        {
+            var pipe = new Pipe();
+            var writer = pipe.Writer;
+            for (var i = 0; i < 3; i++)
+            {
+                var writeOnlyStream = new WriteOnlyPipeStream(writer);
+                writer = new StreamPipeWriter(writeOnlyStream);
+            }
+
+            await writer.WriteAsync(new byte[10]);
+
+            var res = await pipe.Reader.ReadAsync();
+            Assert.Equal(new byte[10], res.Buffer.ToArray());
+        }
+
+        // have to check other internleaved combo
```

Comment:
Will resolve this shortly, wanted to have tests to check Stream APIs that I forgot to add